### ISSUE TITLE
cannot deploy multiple instances for one verticle programly from clojure verticle

### ISSUE THREAD

**Issue by jianglibo** [vert-x/mod-lang-clojure#130]
when deploy one instance, it always work.
when deploy more than one, error occur.
```
Stopping clojure verticle: cn/intellijoy/clojure/file_server.clj
Unexpected exception in Clojure verticle
clojure.lang.ArityException: Wrong number of args (2) passed to: buffer$def-get,
 compiling:(buffer.clj:154)
        at clojure.lang.Compiler$InvokeExpr.eval(Compiler.java:3387)
        at clojure.lang.Compiler.eval(Compiler.java:6516)
        at clojure.lang.Compiler.load(Compiler.java:6952)
        at clojure.lang.RT.loadResourceScript(RT.java:359)
        at clojure.lang.RT.loadResourceScript(RT.java:350)
        at clojure.lang.RT.load(RT.java:429)
        at clojure.lang.RT.load(RT.java:400)
        at clojure.core$load$fn__4890.invoke(core.clj:5415)
        at clojure.core$load.doInvoke(core.clj:5414)
        at clojure.lang.RestFn.invoke(RestFn.java:408)
        at clojure.core$load_one.invoke(core.clj:5227)
        at clojure.core$load_lib.doInvoke(core.clj:5264)
        at clojure.lang.RestFn.applyTo(RestFn.java:142)
```

I think when vertx launch instance, will load vertx.buffer more than once, that cause the problem.

**Comment by tobias** [comment]
This issue has been reported before, but I've never been able to recreate it. Would you be willing to provide a small project that recreates it?

**Comment by jianglibo** [comment]
ok.
1.checkout https://github.com/jianglibo/filetransfer
2.into project folder, mvn install
3.run vertx runzip target\filetransfer-1.0-SNAPSHOT-mod.zip -conf bm_s.json or bm_c.json for client.
4.edit bm_s.json,change instances number.

when I setup more than 10 instances,it always cause error. when less than 3 instances, sometimes work.

**Comment by jianglibo** [comment]
E:\bms>java -version
java version "1.7.0_60"
Java(TM) SE Runtime Environment (build 1.7.0_60-b19)
Java HotSpot(TM) 64-Bit Server VM (build 24.60-b09, mixed mode)

E:\bms>vertx version
2.1.5 (built 2014-11-13 15:15:56)

**Comment by jianglibo** [comment]
OS is windows7.

I test this code in centos7 virtual machine, it just works.has no problem.

**Comment by jianglibo** [comment]
when i trace into code, I find follwing code:
```
in vertx.core:

(defn- -start-verticle [main id]
  (binding [-current-verticle-id id]
    (clojure.lang.RT/loadResourceScript main)))

in ClojureVerticleFactory:

public void start() {
    this.runtime.invoke("vertx.core/-start-verticle", scriptName, this.id);
    log.info("Started clojure verticle: " + scriptName);
}
```

why use clojure.lang.RT/loadResourceScript, but not use clojure's require load etc.

**Comment by jianglibo** [comment]
*deploy clojure verticle has concurrency problem*

if run clojure.lang.RT/loadResourceScript for same file very quickly, it's may have concurrency problem, I guess.

when you deploy multiple instance, some namespace resolve problem will occur. So I change the starter verticle, deploy verticle instance one by one. No error occur.

```
(defn deploy-server-one [cfg inst-num]
  (deploy-verticle "cn/intellijoy/clojure/file_server.clj"
                       :config cfg
                       :instances 1
                       :handler (fn [err deploy-id]
                                  (when err
                                    (log/info "deploy file server failed."))
                                  (swap! inst-num - 1)
                                  (when (> @inst-num 0)
                                    (deploy-server-one cfg inst-num)))))

```

inst-num is the number of instance you want to deploy.

**Comment by tobias** [comment]
Thanks for digging deeper on this issue! I just released version 1.0.5-SNAPSHOT of the language plugin that should fix the concurrent loading problems. Can you confirm it works for you? You'll need to change the clojure module version in your `langs.properties`.

We use `RT/loadResourceScript` instead of `clojure.core/load` because `loadResourceScript` is designed to load full script names, which is how vertx verticles are typically specified. I would be open to the idea of supporting loading verticles by namespace name in addition to path to the .clj. Please open another issue if you would like to have that.

**Comment by jianglibo** [comment]
Yes! It works. I start 100 instance, It's fine. Thank you very much.
